dolfyn/adp/turbulence.py

```
@@ -0,0 +1,753 @@
+                   'long_name': 'Doppler Noise Level',
+                   'description': 'Doppler noise level calculated '
+                   'from PSD white noise'})
+        return out
+
+    def calc_stress_4beam(self, ds, noise=None, orientation=None, beam_angle=25):
+        """Calculate the stresses from the covariance of along-beam 
+        velocity measurements
+
+        Parameters
+        ----------
+        ds : xarray.Dataset
+          Raw dataset in beam coordinates
+        noise : int or xarray.DataArray (time)
+          Doppler noise level in units of m/s
+        orientation : str, default=ds.orientation
+          Direction ADCP is facing ('up' or 'down')
+        beam_angle : int, default=25
+          ADCP beam angle in units of degrees
+
+        Returns
+        -------
+        stress_vec : xarray.DataArray(s)
+          Stress vector with u'w'_ and v'w'_ components
+
+        Notes
+        -----
+        Assumes zero mean pitch and roll.
+
+        Assumes ADCP instrument coordinate system is aligned with principal flow
+        directions.
+
+        Stacey, Mark T., Stephen G. Monismith, and Jon R. Burau. "Measurements 
+        of Reynolds stress profiles in unstratified tidal flow." Journal of 
+        Geophysical Research: Oceans 104.C5 (1999): 10933-10949.
+        """
+
+        warnings.warn("    The 4-beam stress equations assume the instrument's "
+                      "(XYZ) coordinate system is aligned with the principal "
+                      "flow directions.")
+        if noise is None:
+            noise = 0
+            warnings.warn(
+                '    No "noise" input supplied. Consider calculating "noise" using `calc_doppler_noise`')
+
+        if 'beam' not in ds.coord_sys:
+            warnings.warn("    Raw dataset must be in the 'beam' coordinate system. "
+                          "Rotating raw dataset...")
+            ds.velds.rotate2('beam')
+
+        b_angle = getattr(ds, 'beam_angle', beam_angle)
```

See comment below about using a silent default here.